expression/builtin_string.go

```diff
@@ -799,49 +799,38 @@ func (c *convertFunctionClass) getFunction(ctx context.Context, args []Expressio
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinConvertSig{newBaseBuiltinFunc(args, ctx)}
+	bf := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpString)
```

Comment:
we always set charset and collate to utf8 in convert?

``` sql
mysql> select convert("123" using binary);
Field   1:  `convert("123" using binary)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  binary (63)
Length:     3
Max_length: 3
Decimals:   31
Flags:      BINARY


+-----------------------------+
| convert("123" using binary) |
+-----------------------------+
| 123                         |
+-----------------------------+
1 row in set (0.00 sec)
```

Reply:
see #4436

Reply:
I can't get what you mean?

Reply:
Currently, the behavior of function `CONVERT` in TiDB is different from MySQL:
1. As #4436 states: `charset` parameter should be a constant. In `getFunction()` we should set the charset and flag using the constant `charset` parameter.
2. Charsets supported by TiDB may not supported in MySQL, like "866"

Reply:
Is convert to binary supported?

Reply:
not, it's another issue: Charsets supported by MySQL may not supported in TiDB, like "binary"

Reply:
Ok

Reply:
I think a TODO is needed here since after supporting binary charset, we also need to add a binary flag here.

Reply:
Rest LGTM

Reply:
added, PTAL